tss-esapi/src/interface_types/session_handles.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::{
+    constants::SessionType, handles::SessionHandle, interface_types::algorithm::HashingAlgorithm,
+    Error, Result, WrapperErrorKind,
+};
+use std::convert::TryFrom;
+
+/// Enum representing an policy session interface type
+///
+/// # Details
+/// This corresponds to TPMI_SH_POLICY but provides more
+/// information regarding the parameters used when the policy session
+/// was created.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum PolicySession {
+    PolicySession {
```

Comment:
Still think it's a bit superfluous, but I'm ok with this as the final form

Reply:
I actually agree with you and I was going back and forth with different approaches. This approach had to me two advantages:
1. It makes the interface types consistent (they are all enums)
2. And it makes them resemble how they are defined in the specification.